src/software/ai/hl/stp/tactic/pass_defender/pass_defender_tactic_test.py

```diff
@@ -83,5 +83,102 @@ def test_ball_chipped_on_intercept(
     )
 
 
+@pytest.mark.parametrize(
+    "ball_initial_position,ball_initial_velocity,position_to_block_from",
+    [
+        # Place defender behind the ball and direct ball away from defender.
+        # Defender should avoid/abort intercepting the ball in these scenarios.
+        #  <---------O         X
+        #           ball    defender
+        (tbots.Point(-1, 0), tbots.Vector(-3, 0), tbots.Point(0, 0)),
+        #  <---------O    ball
+        #
+        #            X    defender
+        (tbots.Point(0, 0), tbots.Vector(-3, 0), tbots.Point(0, -1)),
+        #            ^
+        #            |
+        #            |
+        #            O    ball
+        #
+        #            X    defender
+        (tbots.Point(0, 1), tbots.Vector(0, 3), tbots.Point(0, 0)),
+    ],
+)
+def test_avoid_intercept_scenario(
```

Comment:
For some reason the new tests crash for me without thunderscope, and with thunderscope enabled, it seems like it freezes. Have you noticed this as well?

Reply:
Tests run fine and pass for me, except without tscope some tests fail (#2850, #2858). Maybe someone else could confirm whether these tests are freezing up for them?

Reply:
Gotcha, I debugged and found that there's a race condition. I've added more info to #2850